### PR TITLE
Add Papers v3.4.20

### PR DESCRIPTION
**Pull request by chgallen**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask-versions/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

Papers3 is last version of popular reference-manager software Papers released without a license by subscription, before acquisition by and merger with ReadCube. Papers3 still supported and available for download.

### CODE REVIEW

File: Casks/papers3.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+cask 'papers3' do
+  version '3.4.20_579'
+  sha256 '48882e7852cd60ee42356e16fc297bbbe4f49c6a01f8a4a9127cfd10e45ff775'
+
+  # downloads.mekentosj.com was verified as official when first introduced to the cask
+  url "https://downloads.mekentosj.com/papers_#{version.no_dots}.dmg"
```
Comment (vitorgalvao) [inline Casks/papers3.rb:6]: Where’s this?

File: Casks/papers3.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  name 'Papers3'
```
Comment (vitorgalvao) [inline Casks/papers3.rb:7]: Remove the number. Make it the same as the main cask.

File: Casks/papers3.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  homepage 'https://support.papersapp.com/support/solutions/30000034695'
+
+  app 'Papers.app'
+
+  uninstall login_item: 'Citations'
+
+  zap trash: [
```
Comment (vitorgalvao) [inline Casks/papers3.rb:14]: Replace the `3`s with `#{version.major}`.

File: Casks/papers3.rb (hunk 1 above)
Comment (chgallen) [reply Casks/papers3.rb:6]: Mekentosj was the developer of Papers before it was merged into ReadCube. mekentosj.com is now a redirect to papersapp.com.

File: Casks/papers3.rb (hunk 2 above)
Comment (chgallen) [reply Casks/papers3.rb]: Done ✅

File: Casks/papers3.rb (hunk 3 above)
Comment (chgallen) [reply Casks/papers3.rb:14]: Done ✅

File: Casks/papers3.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/papers3.rb:6]: That’s not good enough, I’m afraid. We need a link that goes from the homepage to `url` in a way anyone can confirm it’s official without having inside knowledge.

For all a user knows, mekentosj.com may be redirecting now but point to something malicious later.

Comment (chgallen) [reply Casks/papers3.rb:6]: Oh sorry, misunderstood what you needed.

Link from papersapp.com to the legacy download at mekentosj.com is here: https://support.papersapp.com/support/solutions/articles/30000031865-existing-papers-3-users-accessing-papers-3-program-files-for-additional-device-installs

**Pull request merged**